### DISCUSSION TITLE
Disable VPN temporarily?

### DISCUSSION THREAD

**Discussion by klesky415** [haugene/docker-transmission-openvpn#2143] [Q&A]
Hi Team,

Is it possible to disable the VPN, via a command? As it is something that would be useful to do for testing purposes.

Thanks in advance

**Comment by pkishino** [comment]
Please outline a useful use-case for this? There previously was some discussion around this but nothing more. I don’t personally see any point in this so please elaborate

**Comment by klesky415** [comment]
I'm finding that at times I have issues with the VPN provider, be it from the provider's routers not doing what they're supposed to do. Or potentially the VPN servers being blocked. If I run the same torrent on my home connection, it downloads with no issues whatsoever. As with all great issues it's intermittent.

The other case be it related to above or not, is if that you're looking for some obscure low seed content, you tend to get it with running on my home connection, whilst the VPN one can never ever connect.

NordVPN is the provider for me.

My current workaround, is to stop this docker container, and boot up another one that has transmission with no VPN, and shares the rest of the config.

**Comment by nullset2** [comment]
A use case is that the user may need to tunnel in a different way. For example, self-managed at the router level, instead of at the container level, to not depend on the script.

I know, of course, that other images exist around which run transmission only, but that involves migrating the config over; for the sake of convenience, an env variable or some other configuration could be provided which would determine if the openvpn script should be run or not. I consider it a very small surface area change.

**Comment by edgd1er** [comment]
Why not use a simple transmission container like https://docs.linuxserver.io/images/docker-transmission/, provided the transmission's version is the same, /config and /data dir could be shared between the two containers, not running at the same time though.
You could even have the same docker-compose file and comment/uncomment the images.

Having a variable to enable/disable the vpn is as far as I'm concerned, is a risk too big. 
at the moment, transmission won't be started if vpn is not up and on down event the client is stopped, assuring transmission client to go through vpn. default route are removed to be even more sure.
As usual you can't have security, usability and functionality at 100%. https://blog.c3l-security.com/2019/06/balancing-functionality-usability-and.html
